From f5a0c133e91a3892e86c09b1e2a428abac0b6c54 Mon Sep 17 00:00:00 2001
From: Yongjoo Ahn <yongjoo1.ahn@samsung.com>
Date: Fri, 16 Oct 2020 16:56:51 +0900
Subject: [PATCH] Patch for yocto build

- Fix build script to resolve openblas and iniparser
- Mark jsoncpp and curl to be not required

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>
---
 meson.build | 33 +++++++++++++++++++++++----------
 1 file changed, 23 insertions(+), 10 deletions(-)

diff --git a/meson.build b/meson.build
index 91e5e6d..eb035bb 100644
--- a/meson.build
+++ b/meson.build
@@ -94,11 +94,7 @@ endif
 
 if get_option('enable-blas')
   add_project_arguments('-DUSE_BLAS=1', language:['c','cpp'])
-  if build_platform == 'tizen'
-    blas_dep = dependency('openblas')
-  else
-    blas_dep = dependency('blas-openblas')
-  endif
+  blas_dep = dependency('openblas')
 endif
 
 if get_option('use_gym')
@@ -116,19 +112,36 @@ endif
 libm_dep = cxx.find_library('m') # cmath library
 libdl_dep = cxx.find_library('dl') # DL library
 thread_dep = dependency('threads') # pthread for tensorflow-lite
-iniparser_dep = dependency('iniparser', required : false) # iniparser
+
+sysroot = run_command(
+    cxx.cmd_array() + ['-print-sysroot']
+    ).stdout().split('\n')[0]
+
+if sysroot.startswith('/')
+  sysroot_inc_cflags_template = '-I@0@/usr/include@1@'
+  sysroot_inc = sysroot_inc_cflags_template.format(sysroot, '')
+  add_project_arguments(sysroot_inc, language: ['c', 'cpp'])
+  sysroot_inc_cflags_iniparser = sysroot_inc_cflags_template.format(sysroot,
+      '/iniparser')
+else
+  sysroot_inc_cflags_iniparser = '-I/usr/include/iniparser'
+endif
+
+iniparser_dep = dependency('iniparser', required: false) # iniparser
 if not iniparser_dep.found()
   message('falling back to find libiniparser library and header files')
   libiniparser_dep = cxx.find_library('iniparser')
   if libiniparser_dep.found() and cxx.has_header('iniparser.h', \
-        args : '-I/usr/include/iniparser')
+        args : sysroot_inc_cflags_iniparser)
     iniparser_dep = declare_dependency (dependencies : libiniparser_dep,
-      compile_args : '-I/usr/include/iniparser')
+      compile_args : sysroot_inc_cflags_iniparser)
+  else
+      error('Failed to resolve dependency on iniparser')
   endif
 endif
 
-jsoncpp_dep = dependency('jsoncpp') # jsoncpp
-libcurl_dep = dependency('libcurl')
+jsoncpp_dep = dependency('jsoncpp', required: false) # jsoncpp
+libcurl_dep = dependency('libcurl', required: false)
 gtest_dep = dependency('gtest', required: false)
 
 # Install .pc
-- 
2.17.1

